### PR TITLE
[Fix #4794] Fix parsing of code with a ternary expression in MultilineOperationIndentation

### LINKED ISSUE #4794: Error with multiline ternary operators
## Expected behavior

Describe here how you expected RuboCop to behave in this particular situation.

With this sample code:

```ruby
one ||
  two ? 3 : 5
```

Rubocop should report the offence with multi-line ternary operators according to the configuration.

## Actual behavior

The offence is reported correctly but with an error:

```
1 error occurred:
An error occurred while Layout/MultilineOperationIndentation cop was inspecting /Users/jrmhaig/workspace/rubocop_test/test.rb:1:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
Finished in 0.12138300016522408 seconds
```

## Steps to reproduce the problem

Create a file `test.rb` containing:

```ruby
one ||
  two ? 3 : 5
```

and execute Rubocop.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

### PR DESCRIPTION
**Pull request by rrosenblum**
This fixes #4794. The nodes that are returned by `kw_node_with_special_indentation` are expected to have a `keyword`. Ternary expressions do not have a keyword so it seemed best to ignore them. There might be some opportunity to improve upon this later.

### CODE REVIEW

File: lib/rubocop/cop/mixin/multiline_expression_indentation.rb [hunk 1]
```diff
@@ -109,6 +109,7 @@ def keyword_message_tail(node)
       def kw_node_with_special_indentation(node)
         keyword_node =
           node.each_ancestor(*KEYWORD_ANCESTOR_TYPES).find do |ancestor|
+            next if ancestor.respond_to?(:ternary?) && ancestor.ternary?
```
Comment (Drenmi) [inline lib/rubocop/cop/mixin/multiline_expression_indentation.rb]: You could replace for:

```ruby
next if ancestor.if_type? && ancestor.ternary?
```

to get rid of the pesky `#respond_to?`.

**Comment by Drenmi** [comment]
Yep. I agree with your assessment. It was my conclusion as well.

**Comment by Drenmi** [comment]
You'll also need to regenerate the TODO, since the longest module got one line longer. 🙂

**Comment by rrosenblum** [comment]
Updated the code and the TODO.

**Comment by bbatsov** [comment]
This has to be rebased on top of the current `master`.

**Comment by rrosenblum** [comment]
Rebased

**Pull request merged**